### ISSUE TITLE
Hexagon (HTP) backend: prefill produces garbage logits for ubatch ≥ 32 tokens (Qwen3, Hexagon v79)

### ISSUE THREAD

**Issue by bnifullrate-rgb** [ggml-org/llama.cpp#24201]
### Summary

On the Hexagon NPU backend (`-DGGML_HEXAGON=ON`, `--device HTP0`), **prefill produces corrupted
logits when a ubatch contains ≥ 32 tokens** — generation is garbage from the very first sampled
token (e.g. `100 3 33 33 33 …`, mixed-script junk). With `--ubatch-size ≤ 16` the same prompt
produces correct, coherent output. The split logic is fine; the corruption is in the HTP compute
kernel's handling of the token (batch) dimension.

### Environment

- **Device:** OnePlus 13 — Snapdragon 8 Elite, **Hexagon arch v79** (`libggml-htp-v79.so`)
- **Backend:** `-DGGML_HEXAGON=ON -DGGML_OPENCL=ON`, built with the `snapdragon-toolchain`
  Docker image (`ghcr.io/snapdragon-toolchain/arm64-android:v0.7`, bundles Hexagon SDK 6.6 + NDK r28b)
- **llama.cpp:** tag `b9505` (commit `0066404`)
- **Model:** Qwen3.5-4B fine-tune, **Q4_0** (also reproduces on a 9B Q4_0)
- `GGML_HEXAGON_NDEV=1`, single HTP session

### Reproduction

```sh
# GARBAGE — first token onward is junk:
GGML_HEXAGON_NDEV=1 ./bin/llama-completion --device HTP0 -ngl 99 -m model-Q4_0.gguf \
  -p "The sun rose over the quiet valley. Birds sang in the tall pine trees. A farmer walked toward his field carrying a wooden bucket. The morning air was cool and smelled of grass. He paused to look at the distant mountains. Then he continued on his way." \
  -n 32 -c 4096 -ub 32 --temp 0

# COHERENT — same prompt, smaller ubatch:
... same command with -ub 16
```

Measured (binary-searched on a 150-token prompt): **ub ≤ 31 → coherent; ub = 32 → garbage.** The
break is **exactly at 32**, which equals one full HVX vector on v79 (1024-bit = 32 × fp32). A short
prompt (single ubatch < 32 tokens) is always correct, which is why trivial prompts appear to "work"
while any multi-sentence prompt corrupts.

### Likely cause (a lead, not a confirmed fix)

The corruption boundary lands exactly at **32 tokens along the batch dimension**, and HVX vectors on
Hexagon v79 are **1024 bits = 128 bytes = 32 × fp32**. This strongly suggests an **HVX
vectorization bug at the one-full-vector boundary** in the HTP matmul kernel (`ggml/src/ggml-hexagon/htp/`)
— correct for a partial vector (< 32 elements along the token axis), wrong once a full HVX vector (or
multiple vectors) is processed. `test-backend-ops -b HTP0 -o MUL_MAT` passes for tiny shapes (n ≤ 3),
which wouldn't exercise the ≥ 32 batch path.

### Workaround

Setting `n_ubatch = 31` (one element below the 32-element HVX boundary) gives correct output at some
prefill-throughput cost. We'd love a kernel-level fix so full-size prefill batches are correct.

Happy to provide more repro detail, additional model/arch combos, or `GGML_HEXAGON_VERBOSE` /`GGML_HEXAGON_PROFILE` dumps.
